src/Enchantments.h

```diff
@@ -139,8 +142,9 @@ class cEnchantments
 	/** Check enchantment conflicts from enchantments from the vector */
 	static void CheckEnchantmentConflictsFromVector(cWeightedEnchantments & a_Enchantments, const cEnchantments & a_FirstEnchantment);
 
-	/** Gets random enchantment from Vector and returns it */
-	static cEnchantments GetRandomEnchantmentFromVector(const cWeightedEnchantments & a_Enchantments);
+	/** Gets random enchantment from Vector and returns it
+	Passing an MTRand PRNG pointer is optional. */
+	static cEnchantments GetRandomEnchantmentFromVector(const cWeightedEnchantments & a_Enchantments, MTRand * Random = nullptr);
```

Comment:
Are we happy passing `MTRand` like this?

Reply:
Usually GetRandomProvider() Is used

```auto Random = GetRandomProvider(); ```

This accesible from anywehre

Reply:
Sure but make it MTRand &

Reply:
I used a pointer to indicate it was optional. I'd heard it was convention to pass a pointer if it could be nullptr, and a reference if it had to exist.

Reply:
In #4842 madmaxoft said:

> As for by-pointer vs by-reference vs by-value:
Stuff passed by-pointer means that it may be optional (nullptr). When you pass by-reference, you maek it clear to the caller that the parameter must be valid. Filling with loot requires a valid world, so we make it clear by passing by-reference. By-value passing is used for tiny stuff that is cheap to copy (can fit into a few registers) - basically plain numbers and Vector3<> instances were decided to be small enough to always pass by-value.

Reply:
You're passing in a MTRand for every invocation of GetRandomEnchantmentetc right? Or have I misread.

Reply:
I am for everything related with enchantments so I can make sure the result will always be the same since I'm passing the same PRNG state. However maybe there'd be use cases where you don't care about the PRNG state or reproducibility (maybe for chest loot or plugin calls).

Reply:
The more I think, the more I realise I've been overcomplicating this. It's not much to expect the caller to call GetRandomSource() for us if needs be.